### ISSUE TITLE
Copy Region blocks do not appear when individual block visibility options set

### ISSUE THREAD

**Issue by kelizoliva** [backdrop-contrib/copy_blocks#18]
I am attempting to copy a region with 5 blocks in one layout into another layout. I made a copy of the region, selecting the correct layout and region, however the block itself on the front end is empty.
![Screen Shot 2021-04-01 at 8 16 21 AM](https://user-images.githubusercontent.com/44212945/113315910-a5542d00-92c2-11eb-939d-365e2a285a7f.png)

![Screen Shot 2021-04-01 at 8 17 56 AM](https://user-images.githubusercontent.com/44212945/113316081-d2084480-92c2-11eb-9200-005eeedf4716.png)

This is the case for any regions I attempt to copy. Copying individual blocks works without issue. The blocks in the region to be copied are just custom blocks, however all of the blocks in the regions I am attempting to copy have visibility options set.  The visibility options for these blocks are by user role.

As a control I dropped in the powered by backdrop block with no visibility conditions set, and it does show up in the copied region. When I set the same visibility conditions on that block as the others, it no longer shows up in the copied block.

Issue #7 seems to have addressed visibility options not being respected, but it seems there is now something else afoot here.

**Comment by kelizoliva** [comment]
I see now that in #9 @laryn mentions that the merged code has revealed this issue.

**Comment by laryn** [comment]
Agreed, visibility conditions don't seem to be working in copied regions. I am seeing a lot of blocks showing up that should not show up (which sounds opposite of what you are seeing) but haven't been able to track down where the issue is yet.

**Comment by laryn** [comment]
Oops, sorry for the noise. The site I was seeing that on was running an old version of Copy Blocks. I updated to the latest release and now the visibility conditions are being respected (in this case it is a language visibility condition -- certain fields only show for certain languages).

@kelizoliva Can you verify that it is specifically the user role visibility condition that is not working for you?

**Comment by kelizoliva** [comment]
> @kelizoliva Can you verify that it is specifically the user role visibility condition that is not working for you?

Yes, this is for role-based visibility, though I have not actually tested other types of visibility conditions. This may be limited to role conditions, but might extend to other conditions as well.

**Comment by jenlampton** [comment]
I can confirm the role based visibility does not seem to work across layouts. I have my source block on the default layout, and on other layouts I get the following fatals:

```
Warning: Undefined array key "user" in UserRoleLayoutAccess->checkAccess() (line 38 of /var/www/html/web/core/modules/layout/plugins/access/user_role_layout_access.inc).
Warning: Attempt to read property "data" on null in UserRoleLayoutAccess->checkAccess() (line 38 of /var/www/html/web/core/modules/layout/plugins/access/user_role_layout_access.inc).
Warning: Attempt to read property "roles" on null in UserRoleLayoutAccess->checkAccess() (line 39 of /var/www/html/web/core/modules/layout/plugins/access/user_role_layout_access.inc).
TypeError: array_intersect(): Argument #2 must be of type array, null given in array_intersect() (line 40 of /var/www/html/web/core/modules/layout/plugins/access/user_role_layout_access.inc).
```

These errors show up for the editor (but not for the developer / admin role)

I have one block that's only visible to editors, and another thats for all role's other than editor (I used the "negate" option) placed in the default layout's sidebar region. This region is copied to all other sidebars using copy_blocks.

edit: I tried removing conditions from first one block, then the other, but the same 4 errors remained with all 3 scenarios. removing all conditions restored my site to normal.

**Comment by jenlampton** [comment]
I just tried to prevent my blocks from apearing using `mmsutil_block_view_alter`, but that also does not affect blocks that are rendered through copied regions in copy_blocks.
